src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
 'use strict';
 
-// write code here
+// TABLE VARIABLES
+const table = document.querySelector('table');
+const tHead = table.tHead;
+const tBody = table.tBodies[0];
+
+// SORTING FUNCTIONS AND RELATED OBJECTS
+const valueByIndex = (row, index) => row.cells[index].textContent;
+const salaryToNum = (a) => Number(a.substring(1).split(',').join(''));
+
+const getSortingFunction = (index, order, transformFunc = (x) => x) => {
+  return (a, b) => {
+    const valueA = transformFunc(valueByIndex(a, index));
+    const valueB = transformFunc(valueByIndex(b, index));
+
+    if (typeof valueA === 'string' && typeof valueB === 'string') {
+      return order * valueA.localeCompare(valueB);
+    }
+
+    return order * (valueA - valueB);
+  };
+};
+
+const sortingFunctions = {
+  transformFunctions: {
+    age: parseInt,
+    salary: salaryToNum,
+  },
+
+  get: function (index, colHeader, ascOrder) {
+    const transformFunc = this.transformFunctions[colHeader];
+
+    return ascOrder
+      ? getSortingFunction(index, 1, transformFunc)
+      : getSortingFunction(index, -1, transformFunc);
+  },
+};
+
+// OBJECT TO KEEP TRACK OF ASC/DESC ORDER
+const orderByUtil = {
+  element: undefined,
+  ascOrder: undefined, // true -> asc, false -> desc
+
+  clickIsMade: (target) => {
+    if (target === orderByUtil.element) {
+      orderByUtil.ascOrder = !orderByUtil.ascOrder;
+    } else {
+      orderByUtil.element = target;
+      orderByUtil.ascOrder = true;
+    }
+  },
+  getValue: () => orderByUtil.ascOrder,
+};
```

Comment:
The `orderByUtil` object is mutable, which might lead to bugs in larger codebases. If possible, consider refactoring your code to avoid mutating global objects.